### PR TITLE
[SignalR] Avoid Http2 when negotiate auth is used

### LINKED ISSUE #45371: SignalR Client Windows Authentication returning 401Unautorized (update 6.0.11 -> 7.0.0)
### Description

After updating the version of Microsoft.AspNetCore.SignalR.Client from 6.0.11 to 7.0.0 I receive a 401 (Unauthorized) when trying to start the connection using default credentials. I couldn't find any information about potential breaking changes.

```
  connection = new HubConnectionBuilder()
  .WithUrl(hubUrl, options => {
      options.UseDefaultCredentials = true;
  }).WithAutomaticReconnect().Build();
    
   await connection.StartAsync(); ---> Response status code does not indicate success: 401 (Unauthorized).
```

### Configuration
VS2022
.NET 7
Windows 10
x64
All Browsers

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/dotnet/aspnetcore/issues/45371

https://github.com/dotnet/aspnetcore/blob/c096dbbbe652f03be926502d790eb499682eea13/src/Security/Authentication/Negotiate/src/NegotiateHandler.cs#L54

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

File: src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs [hunk 1]
```diff
@@ -542,6 +538,7 @@ private HttpClient CreateHttpClient()
         HttpMessageHandler httpMessageHandler = httpClientHandler;
 
         var isBrowser = OperatingSystem.IsBrowser();
+        var useHttp2 = true;
```
Comment (halter73) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs:541]: Nit: Can we call this `allowHttp2` like you do in the WebSocketTransport?

**Review by halter73 — APPROVED** [review]
It's nice not having to set `HttpRequestMessage.Version` everywhere.

**Comment by BrennanConroy** [comment]
> It's nice not having to set HttpRequestMessage.Version everywhere.

Yeah, not sure why we didn't do it that way to start with

File: src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs [hunk 2]
```diff
@@ -1788,6 +1788,110 @@ public async Task WebSocketsCanConnectOverHttp2()
         Assert.Contains(TestSink.Writes, context => context.Message.Contains("Request finished HTTP/2 CONNECT"));
     }
 
+    [Theory]
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs:1791]:
```suggestion
    [ConditionalTheory]
```

**Pull request merged**